builtin/providers/aws/json.go

```diff
@@ -0,0 +1,38 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func normalizeJson(jsonString interface{}) string {
+	if jsonString == nil || jsonString == "" {
+		return ""
+	}
+	var j interface{}
+	err := json.Unmarshal([]byte(jsonString.(string)), &j)
+	if err != nil {
+		return fmt.Sprintf("Error parsing JSON: %s", err)
+	}
+	b, err := json.Marshal(j)
+	if err != nil {
+		panic(err) // unexpected
+	}
+	return string(b)
+}
+
+func normalizePolicyDocument(policyString interface{}) string {
+	if policyString == nil || policyString == "" {
+		return ""
+	}
+	var policy IAMPolicyDoc
+	err := json.Unmarshal([]byte(policyString.(string)), &policy)
+	if err != nil {
+		return fmt.Sprintf("Error parsing IAM policy document: %s", err)
```

Comment:
Re: returning an error. This matches the behavior of the existing `normalizeJson` function. I can file a separate ticket to fix both.

Re: `interface{}`. This allows `normalizePolicyDocument` to be used as a [`StateFunc`](https://github.com/hashicorp/terraform/blob/57cf9fd2953192f274731bfe4d38ea4e39e74270/helper/schema/schema.go#L194).